### PR TITLE
Some fixes

### PR DESCRIPTION
**Pull request by peterNordin**
I could not compile on my machine, this PR contains some fixes I had to make.
It also contains some fixes to warnings reported by static code analysis in qtcreator.
But I did not fix all warnings, I can do some additional work if you want.

Also one commit is still WIP, need your input.

### CODE REVIEW

File: include/fmi4c_private.h [hunk 1]
```diff
@@ -27,7 +26,7 @@
 #define LOADFUNCTION2(FUNCNAME) (FUNCNAME ## _t)GetProcAddress(dll, getFunctionName(fmu->fmi1.modelName, #FUNCNAME));
 #define LOADFUNCTION(FUNCNAME) (FUNCNAME ## _t)GetProcAddress(dll, #FUNCNAME);
 #else
-#define LOADFUNCTION2(FUNCNAME) (FUNCNAME ## _t)dlsym(dll, getFunctionName(fmu->modelName, #FUNCNAME));
+#define LOADFUNCTION2(FUNCNAME) (FUNCNAME ## _t)dlsym(dll, getFunctionName(fmu->instanceName, #FUNCNAME));
```
Comment (robbr48) [inline include/fmi4c_private.h:29]: I believe this should be `fmu->fmi1.modelname`. But I removed the load function macros and replaced them with a function anyway.

File: src/fmi4c.c [hunk 2]
```diff
@@ -1408,7 +1410,7 @@ bool loadFunctionsFmi1(fmiHandle *fmu)
     getcwd(cwd, sizeof(char)*FILENAME_MAX);
 #endif
 
-    char *dllPath = (char*)malloc(sizeof(char)*FILENAME_MAX);
+    char dllPath[FILENAME_MAX];
```
Comment (robbr48) [inline src/fmi4c.c:1413]: Not sure why I used malloc here 😕 But I guess the result will be equivalent, just more readable?

**Review by robbr48 — APPROVED** [review]
Looks good! I especially appreciate some help on CMake. There are some minor conflicts due to some of my fixes, but it should be easy to resolve.

File: src/fmi4c.c (hunk 2 above)
Comment (peterNordin) [reply src/fmi4c.c:1413]: Auto cleanup of memory when function return, else memory leek if free is not called manually

**Pull request merged**